modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/SelectionFilter.js

```diff
@@ -92,73 +93,111 @@ function SelectionFilter({
 	id,
 	inputPlaceholder,
 	itemKey,
-	itemLabel: itemLabelProp,
+	itemLabel,
 	items: initialItems,
 	multiple,
 	selectedData,
 	setFilter,
 }) {
-	const [query, setQuery] = useState('');
-	const [search, setSearch] = useState('');
+	const [searchOptions, setSearchOptions] = useState({
```

Comment:
@markocikos Combine `currentPage` & `search` state variables as they usually change together and prevents triggering an extra request to the backend.